scripts/build-ancillary-repository.sh

```diff
@@ -142,7 +142,7 @@ function build_ancillary_repository() {
 #    environment variables, and the script will work as expected.
 #
 
-upstream_branch="${UPSTREAM_PRODUCT_BRANCH:-master}"
+upstream_branch="${UPSTREAM_BRANCH:-master}"
```

Comment:
We do not need this anymore since `UPSTREAM_BRANCH` should always be set.
Can you please get rid of `upstream_branch` and do a fatal check that `UPSTREAM_BRANCH` is set?

Reply:
shellcheck failed for me when I did a check of UPSTREAM_BRANCH (e.g. `if [[ -z UPSTREAM_BRANCH ]];...`) without previously assigning it in the script, so I opted to reassign to `upstream_branch` defaulting to nothing, and then check `upstream_branch`.